File: problems/2d-gps/scaffold.cpp
#include <iostream>

struct Coordinate
{
    double x;
    double y;
    double distance;

    Coordinate(double x, double y, double distance) : x(x), y(y), distance(distance) {}
};

struct Point
{
    double x;
    double y;

    Point(double x, double y) : x(x), y(y) {}
};

Point trilaterate(const Coordinate &p1, const Coordinate &p2, const Coordinate &p3)
{
    // Your solution goes here
    return Point(x, y);
}

Coordinate readCoordinate()
{
    double x, y, distance;
    std::cin >> x >> y >> distance;
    return Coordinate(x, y, distance);
}

int main()
{
    Coordinate p1 = readCoordinate();
    Coordinate p2 = readCoordinate();
    Coordinate p3 = readCoordinate();

    Point result = trilaterate(p1, p2, p3);

    std::cout << std::fixed;
    std::cout << (int)result.x << " " << (int)result.y << std::endl;

    return 0;
}

File: problems/8-2d-gps/scaffold.java
import java.util.Scanner;

class Coordinate {
    public double x;
    public double y;
    public double distance;

    public Coordinate(double x, double y, double distance) {
        this.x = x;
        this.y = y;
        this.distance = distance;
    }
}

class Point {
    public double x;
    public double y;

    public Point(double x, double y) {
        this.x = x;
        this.y = y;
    }
}

public class solution {
    private static Point trilaterate(Coordinate p1, Coordinate p2, Coordinate p3) {
        // Your solution goes here
        return new Point(x, y);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        Coordinate p1 = readCoordinate(scanner);
        Coordinate p2 = readCoordinate(scanner);
        Coordinate p3 = readCoordinate(scanner);

        Point result = trilaterate(p1, p2, p3);

        System.out.println((int)result.x + " " + (int)result.y);
    }

    private static Coordinate readCoordinate(Scanner scanner) {
        double x = scanner.nextDouble();
        double y = scanner.nextDouble();
        double distance = scanner.nextDouble();
        return new Coordinate(x, y, distance);
    }
}